pkgs/development/libraries/srt/default.nix

```diff
@@ -0,0 +1,29 @@
+{ stdenv, fetchFromGitHub, cmake, openssl
+}:
+
+with stdenv.lib;
+stdenv.mkDerivation rec {
+  name = "srt-${version}";
+  version = "1.3.1";
+
+  src = fetchFromGitHub {
+    owner = "Haivision";
+    repo = "srt";
+    rev = "v${version}";
+    sha256 = "0cv73j9c8024p6pg16c4hiryiv4jpgrfj2xhfdaprsikmkdnygmz";
+  };
+
+  nativeBuildInputs = [ cmake ];
+
+  buildInputs = [ openssl ];
+
+  configurePhase = "cmake . -DCMAKE_INSTALL_PREFIX=$out";
```

Comment:
You can use `cmakeFlags = [ "-DCMAKE_INSTALL_PREFIX=${placeholder "out"}" ];`

Reply:
CMAKE_INSTALL_PREFIX should be added automatically by the cmake hook

Reply:
> CMAKE_INSTALL_PREFIX should be added automatically by the cmake hook

@matthewbauer That doesn't work for me, without giving it explicitly, I get this build error:

```
-- Check if compiler accepts -pthread - yes
-- ADDING TRANSITIVE LINK DEP to:srt_shared : -pthread 
-- ADDING TRANSITIVE LINK DEP to:srt_static : -pthread 
-- INSTALL DIRS: bin= lib=/nix/store/vrmb9zrivwnb6519y1ysb8ni7w2j5zdl-srt-1.3.1/lib shlib=/nix/store/vrmb9zrivwnb6519y1ysb8ni7w2j5zdl-srt-1.3.1/lib include=/nix/store/vrmb9zrivwnb6519y1ysb8ni7w2j5zdl-srt-1.3.1/include
-- C++ VERSION: Setting C++11 compat flag for gnu compiler
CMake Error at CMakeLists.txt:636 (install):
  install TARGETS given no RUNTIME DESTINATION for executable target
  "srt-live-transmit".
Call Stack (most recent call first):
  CMakeLists.txt:667 (srt_add_program)
  CMakeLists.txt:690 (srt_add_application)
```

Reply:
I think the idea was not to pass `configurePhase` at all.

Reply:
> You can use `cmakeFlags = [ "-DCMAKE_INSTALL_PREFIX=${placeholder "out"}" ];`

@jtojnar I get the same error as above when trying to do this.

Reply:
> I think the idea was not to pass `configurePhase` at all.

@7c6f434c Yes, that's what I did, and then got that build error.

Reply:
Thanks for clarification; maybe it is because of build dir, no idea.

Reply:
```nix
  cmakeFlags = [
    "-DCMAKE_INSTALL_PREFIX=${placeholder "out"}"
    "-UCMAKE_CXX_COMPILER"
    "-UCMAKE_C_COMPILER"
    "-UCMAKE_AR"
    "-UCMAKE_RANLIB"
    "-UCMAKE_STRIP"
    "-UCMAKE_FIND_FRAMEWORK"
    "-UCMAKE_OSX_SYSROOT"
    "-UCMAKE_OSX_DEPLOYMENT_TARGET"
    "-UCMAKE_POLICY_DEFAULT_CMP0025"
    "-UCMAKE_INSTALL_NAME_DIR"
    "-UCMAKE_INSTALL_LIBDIR"
    "-UCMAKE_INSTALL_INCLUDEDIR"
    "-UCMAKE_BUILD_TYPE"
    "-UCMAKE_SKIP_BUILD_RPATH"
  ];
```

works. One of these is the issue.

Reply:
Yeah, we need to unset `CMAKE_INSTALL_LIBDIR` for some reason.

Reply:
Weird, they disable `GNUInstallDirs` when `CMAKE_INSTALL_LIBDIR` is set

https://github.com/Haivision/srt/blob/6404396588fbc81b6c0f998463da7df479ce31a4/CMakeLists.txt#L35-L37

but `GNUInstallDirs` is what uses `CMAKE_INSTALL_LIBDIR` in the first place

https://cmake.org/cmake/help/v3.0/module/GNUInstallDirs.html

Reply:
I should mention, I got this from other nix packages that do the same, so it seems to be a common problem:

```
% git grep DCMAKE_INSTALL_PREFIX | wc -l
28
```

Reply:
The proper way to package cmake project is to rely on cmake setup hook:

https://github.com/NixOS/nixpkgs/blob/c64951515b325d01c391b83ba0c710fb0d2383f8/pkgs/development/tools/build-managers/cmake/setup-hook.sh

and add project-specific flags through `cmakeFlags`.

Some packagers are probably not aware of this, or the projects use CMake improperly like srt does. Unfortunately, due to CMake’s complexity, this is far too common; I have yet too see a project whose CMake setup is not flawed in some way.

Reply:
Resolving as I've applied @jtojnar's short workaround from https://github.com/NixOS/nixpkgs/pull/54463#discussion_r249921888